### PR TITLE
[assignment 2] Convert version route to plugin

### PR DESCRIPTION
**Pull request by TheAlphaNerd**
Closes #43

### CODE REVIEW

File: lib/version.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Declare internals
+var internals = {
+  response: {
+    version: require('../package.json').version
+  }
+};
+
+// Register plugin
+exports.register = function (server, options, next) {
+    server.route({
```
Comment (AdriVanHoudt) [inline lib/version.js]: newline after function

File: lib/version.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        method: 'GET',
+        path: '/version',
+        config: {
+            description: 'Returns the version of the server',
+            handler: function (request, reply) {
+
+                return reply(internals.response);
+            }
+        }
+    });
```
Comment (AdriVanHoudt) [inline lib/version.js]: I would put a newline here for readability

Comment (TheAlphaNerd) [inline lib/version.js]: done did it

File: lib/version.js (hunk 1 above)
Comment (TheAlphaNerd) [inline lib/version.js]: got it!

File: lib/version.js (hunk 2 above)
Comment (AdriVanHoudt) [inline lib/version.js]: :+1:

File: lib/version.js (hunk 1 above)
Comment (AdriVanHoudt) [inline lib/version.js]: :+1:

File: lib/version.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    next();
```
Comment (hueniverse) [inline lib/version.js]: This is a callback so `return`

Comment (TheAlphaNerd) [inline lib/version.js]: Good call!

**Comment by sschuck** [comment]
Do we want to check if the version is a valid semver?

**Comment by AdriVanHoudt** [comment]
@sschuck the version comes from `package.json` and that should always follow semver

File: lib/index.js [hunk 4]
```diff
@@ -2,31 +2,22 @@
...
 var Hoek = require('hoek');
-var Package = require('../package.json');
-
 
 // Declare internals
 
-var internals = {};
-
+var internals = {
+  version: require('./version')
```
Comment (hueniverse) [inline lib/index.js]: This is odd. Just `var Version = require('./version');` at the top and then use that in `register()`.

Comment (TheAlphaNerd) [inline lib/index.js]: updated

File: lib/index.js [hunk 5]
```diff
@@ -2,35 +2,25 @@
 
 var Hapi = require('hapi');
 var Hoek = require('hoek');
-var Package = require('../package.json');
```
Comment (hueniverse) [inline lib/index.js]: No empty line here.

Comment (TheAlphaNerd) [inline lib/index.js]: In personal projects I have always put a new line between modules that are included from ```node_modules``` and modules that are being included locally from the project.

That aside the style has been fixed.

Comment (AdriVanHoudt) [inline lib/index.js]: I also like to do this, makes it easier to see what are local modules

Comment (hueniverse) [inline lib/index.js]: I just sort them: node, npm, local.

Comment (TheAlphaNerd) [inline lib/index.js]: Reasonable... I follow the same pattern (with spaces).

Thanks for taking the tie to discuss

Comment (AdriVanHoudt) [inline lib/index.js]: same sorting here only with newlines in between

**Pull request merged**